pkg/controller/agent/reconcile.go

```diff
@@ -75,10 +75,13 @@ func reconcilePodVehicle(params Params, podTemplate corev1.PodTemplateSpec) *rec
 		results.WithError(err)
 	}
 
-	err = updateStatus(params, ready, desired)
-	if err != nil && apierrors.IsConflict(err) {
-		params.Logger().V(1).Info("Conflict while updating status")
-		return results.WithResult(reconcile.Result{Requeue: true})
+	err = calculateStatus(&params, ready, desired, status)
+	if err != nil {
+		params.Logger().Error(
+			err, "Error while calculating new status",
+			"namespace", params.Agent.Namespace,
+			"agent_name", params.Agent.Name,
+		)
```

Comment:
As mentioned elsewhere this logged twice once by controller runtime and once here. Consider: 

`pkgerrors.Wrap(err, "while calculating status") `